### PR TITLE
[installer][host] quick hackaround a Sierra >= 10.12.5 issue where di…

### PR DESCRIPTION
**Pull request by srmo**
…skutil -mountPoint no longer works and we need to do same nasty stdout parsing to retrieve the actual mountPoint

### CODE REVIEW

File: installer/host/qt_host_installer/io_osx.cpp [hunk 1]
```diff
@@ -152,12 +152,23 @@ namespace io
...
+          QString actualMountPoint = QString("/Volumes/" + volume);
+          utils::writeLog("mounted " + devicePath + " with message " + stdoutString);
+          utils::writeLog("Assuming mountpoint: " + actualMountPoint);
+          return actualMountPoint;
        }
-       else
-          return true;
    }
 
    bool unmount(QString devicePath, bool isDisk)
```
Comment (samnazarko) [inline installer/host/qt_host_installer/io_osx.cpp]: Should this still be bool if you changed to QString in header?

File: installer/host/qt_host_installer/io.h [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace io
     bool writeImage(QString devicePath, QString deviceImage, QObject* caller=NULL);
     bool unmount(QString devicePath, bool isDisk = false);
     void updateKernelTable();
-    bool mount(QString devicePath, QString mountDir);
```
Comment (samnazarko) [inline installer/host/qt_host_installer/io.h]: This could break things. Use #if defined (Q_OS_MAC) instead.

Comment (srmo) [inline installer/host/qt_host_installer/io.h]: but it's the same interface for linux, I migrated io_linux to follow this interface. would you rather have two distinct implementations?

Comment (srmo) [inline installer/host/qt_host_installer/io.h]: Cool :) I'll change it if it makes any problems.

File: installer/host/qt_host_installer/io_osx.cpp (hunk 1 above)
Comment (srmo) [inline installer/host/qt_host_installer/io_osx.cpp]: I no longer return bool here, either NULL or the new mountpoint

**Comment by samnazarko** [comment]
Had a positive report so far: https://discourse.osmc.tv/t/when-will-os-x-mount-error-solved/37782/11.

Will see if we can get a bit more feedback.

**Comment by srmo** [comment]
yeah, more feedback would be great. Could you verify functionality on windows + linux?

**Comment by samnazarko** [comment]
Seems OK on Windows and Linux, but we don't actually have to build those for now (no changes).

**Comment by samnazarko** [comment]
Positive reports. Thanks again @srmo

**Pull request merged**